lib/usd/utils/MayaTransformAPI.cpp

```
@@ -0,0 +1,1046 @@
+{
+  if(m_api != TransformAPI::kFallback && m_prim)
+  {
+    if(!m_scale)
+      insertScaleOp();
+
+    m_scale.Set(value, time);
+  }
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+GfVec3f MayaTransformAPI::scale(const UsdTimeCode& time) const
+{
+  if(m_api != TransformAPI::kFallback)
+  {
+    if(m_scale)
+    {
+      GfVec3f value;
+      m_scale.Get(&value, time);
+      return value;
+    }
+    return GfVec3f(1.0f);
+  }
+  return _extractScaleFromMatrix(time);
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void MayaTransformAPI::rotate(const GfVec3f& value, const RotationOrder order, const UsdTimeCode& time)
+{
+  const float radToDeg = 180.0f / 3.141592654f;
+  if(m_api != TransformAPI::kFallback && m_prim)
+  {
+    if(!m_rotate)
+    {
+      insertRotateOp(order);
+    }
+
+    m_rotate.Set(value * radToDeg, time);
+  }
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+GfVec3f MayaTransformAPI::rotate(const UsdTimeCode& time) const
+{
+  if(m_api != TransformAPI::kFallback)
+  {
+    const float degToRad = 3.141592654f / 180.0f;
+    if(m_rotate)
+    {
+      GfVec3f value;
+      m_rotate.Get(&value, time);
+      return value * degToRad;
+    }
+    return GfVec3f(0);
+  }
+  return _extractRotateFromMatrix(time);
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+RotationOrder MayaTransformAPI::rotateOrder() const
+{
+  return m_order;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void MayaTransformAPI::translate(const GfVec3d& value, const UsdTimeCode& time)
+{
+  if(m_api != TransformAPI::kFallback && m_prim)
```

Again, we'll get silent failure if m_api != TransformAPI::kFallback.